haas/model.py

```diff
@@ -371,6 +268,49 @@ def get_port_networks(self, ports):
         and have ``session`` just return ``self``.
         """
 
+class Obm(AnonModel):
+    """Obm superclass supporting various drivers 
+
+    related to out of band management of servers.
+    """
+    type = Column(String, nullable=False)
+
+    __mapper_args__ = {
+            'polymorphic_on': type
+            }
+    @staticmethod
+    def validate(kwargs):
+        """Verify that ``kwargs`` is a legal set of keywords args to ``__init__``
+
+        Raise a ``schema.SchemaError`` if the  ``kwargs`` is invalid. 
+        Note well: This is a *static* method; it will be invoked on the class.
+        """     
+        assert False, "Subclasses MUST override the validate method "
+
+    def power_cycle(self):
+        """Power cycles the node.
+
+        Exact implementation to left to the subclasses.
+        """     
+        assert False, "Subclasses MUST override the power_cycle method "
+
+    def start_console(self):
+        """Starts logging to the console. """
+        assert False, "Subclasses MUST override the start_console method" 
+
+    def stop_console(self):
+        """Stops console logging. """
+        assert False, "Subclasses MUST override the stop_console method"
+
+    def delete_console(self):
+        assert False, "Subclasses MUST override the delete_console method"
+
+    def get_console(self):
+        assert False, "Subclasses MUST override the get_console method"
+
+    def get_console_log_filename(self):
+        assert False, "Subclasses MUST override this method"
```

Comment:
Quoting Sahil Tikale (2016-02-25 13:00:19)

> > +    def get_console(self):
> > +        assert False, "Subclasses MUST override the get_console method"
> > +
> > +    def get_console_log_filename(self):
> > +        assert False, "Subclasses MUST override this method"
>
>    which method name you want me to include.
>    I assume it was the power_off method and I included it.
>    It is there (lines 295 - 299 in model.py).
>    Is there anything else I should be including ?

Compare the two methods I quoted: the first (and most all of the others)
say "Subclasses must override the <foo> method".
`get_console_log_filename` however says "Subclasses MUST override this
method." These should be made consistent.